pkg/cableengine/ipsec/strongswan_test.go

```diff
@@ -0,0 +1,66 @@
+package ipsec
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"github.com/kelseyhightower/envconfig"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Util", func() {
+	Describe("Charon port configuration", testCharonConfigPortsGen)
+})
+
+func testCharonConfigPortsGen() {
+	Context("config rendering", func() {
+		It("should render config correctly out of strongSwan parameters", func() {
+			ss := strongSwan{ipSecIKEPort: "500", ipSecNATTPort: "4500"}
+			buf := new(bytes.Buffer)
+
+			ss.renderCharonConfigTemplate(buf)
+
+			Expect(buf.String()).To(ContainSubstring("port = 500"))
+			Expect(buf.String()).To(ContainSubstring("port_nat_t = 4500"))
```

Comment:
and what about the two other options?

Reply:
I'm just testing that the dynamic parts were working correctly.

What's the value of checking that a string in code side A is equal to code side B, and then you need to maintain both?.

In the past I remember using the whole file output as a check to validate config output from code = config output from testing. And that lead to lots of unecesary testing code maintenance.

For those parts of the config file it'd be better to have some sort of functional validation IMHO, although, I haven't even checked what are they meant to be, and we have been running it without them in the right directory (dockerfile was installing it on a wrong place) :P :)

Reply:
The value is to have something check that things break should that change. That is, as long as those options are actually needed.

Reply:
Done